### ISSUE TITLE
CREATE VIEW return incorrect information When a table with the same name exists In hive

### ISSUE THREAD

**Issue by kyo-tom** [trinodb/trino#10037]
With Hive connector
```text
trino:default> CREATE TABLE one (a varchar);
CREATE TABLE
```
CREATE VIEW is rejected when a table with the same name exists,but fail information indicates that the view already exists instead of the table already exists
```text
trino:default> CREATE VIEW one AS SELECT * FROM one;
Query 20211123_061613_00037_jrn8n failed: View already exists: 'default.one'
```

**Comment by kyo-tom** [comment]
cc @sopel39

**Comment by sopel39** [comment]
Message is wrong, but we shouldn't be able to create table anyway

**Comment by kyo-tom** [comment]
> Message is wrong, but we shouldn't be able to create view anyway

I had fix this message in hive,but where should I write a test，is it in trino-product-tests module?

**Comment by sopel39** [comment]
@kyo-tom you can base your test on `io.trino.execution.TestCreateMaterializedViewTask`

**Comment by kyo-tom** [comment]
> io.trino.execution.TestCreateMaterializedViewTask

@sopel39 But it is the behavior of individual catalogs, not all of them

**Comment by sopel39** [comment]
> @sopel39 But it is the behavior of individual plugin, not all of plugin

It should be generic across catalogs

**Comment by kyo-tom** [comment]
> > @sopel39 But it is the behavior of individual plugin, not all of plugin
> 
> It should be generic across catalogs

but some plugin is not support create views. Do you mean creating a MockMetadata

**Comment by sopel39** [comment]
Yes, see how `io.trino.execution.TestCreateMaterializedViewTask` works

**Comment by sopel39** [comment]
Fixed by https://github.com/trinodb/trino/pull/10186

**Comment by findepi** [comment]
https://github.com/trinodb/trino/pull/10186 fixed this on `CreateViewTask` level, but this also should be fixed in individual connectors' metadata classes, to take care of view/table conflict resulting from concurrent creation.
